Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs

```diff
@@ -196,21 +199,59 @@ public QuestionsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
             Assert.True(_trappistDbContext.CodeSnippetQuestionTestCases.Count() == 1);
         }
 
+        [Fact]
+        public async Task UpdateCodeSnippetQuestionAsyncTest()
+        {
+            string userName = "deepankar@promactinfo.com";
+
+            //Adding Application User
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
+
+            //Adding 
+            var codingQuestion = await CreateCodingQuestion();
+            await _questionRepository.AddCodeSnippetQuestionAsync(codingQuestion, applicationUser.Id);
+
+            var question = await _trappistDbContext
+                .Question
+                .FirstOrDefaultAsync(x => x.QuestionDetail == codingQuestion.Question.QuestionDetail);
+
+            codingQuestion.CodeSnippetQuestion.CheckCodeComplexity = false;
+            codingQuestion.CodeSnippetQuestion.CheckTimeComplexity = false;
+            codingQuestion.Question.QuestionDetail = "Updated question details";
+            codingQuestion.Question.DifficultyLevel = DifficultyLevel.Hard;
+            codingQuestion.CodeSnippetQuestion.LanguageList = new string[] { "C" };
+            await _questionRepository.UpdateCodeSnippetQuestionAsync(question.Id, codingQuestion, applicationUser.Id);
+
+            var updatedQuestion = await _trappistDbContext
+                .Question
+                .Include(x => x.CodeSnippetQuestion)
+                .ThenInclude(x => x.QuestionLanguangeMapping)
+                .FirstOrDefaultAsync(x => x.Id == question.Id);
+
+            Assert.True(string.Equals(updatedQuestion.QuestionDetail, codingQuestion.Question.QuestionDetail, StringComparison.CurrentCultureIgnoreCase));
+            Assert.True(updatedQuestion.DifficultyLevel == codingQuestion.Question.DifficultyLevel);
+            Assert.True(updatedQuestion.CodeSnippetQuestion.CheckCodeComplexity == codingQuestion.CodeSnippetQuestion.CheckCodeComplexity);
+            Assert.True(updatedQuestion.CodeSnippetQuestion.CheckTimeComplexity == codingQuestion.CodeSnippetQuestion.CheckTimeComplexity);
```

Comment:
same as above. No difference found.
Please check all the properties and objects that are being updated.